### PR TITLE
RSESPRT-36: Fix Case Tokens icons

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
The Case Token + and - Icons was not working on RSE site. This PR fixes that issue.
When the email activity form is opened for a case, the token list is not properly displayed as one of the token category displays “Undefined“  instead of the token category.

## Before
2021-04-12 at 6 08 PM
image-20210409-114928

## After
2021-04-12 at 6 07 PM

Manage Applications  RSE 2021-04-13 11-38-21 (1)

## Technical Details FE
RSE site is using FontAwesome v5(needed in SSP), but `fa-minus-square-o` and `fa-plus-square-o` classes were not supported on v5. Hence removed `-o`, which is compatible with both v4 and v5.

## Technical Details BE
- The logic for initializing the Email Recipient Tokens [here](https://github.com/compucorp/uk.co.compucorp.civicase/blob/master/CRM/Civicase/Hook/BuildForm/TokenTree.php#L324) checks if the the token tree contains the `Email Recipient` key before it is intialized and the core field tokens added. However for this client, some custom fields for the contact has already been added to the `newTokenTree` variable [here](https://github.com/compucorp/uk.co.compucorp.civicase/blob/master/CRM/Civicase/Hook/BuildForm/TokenTree.php#L95) and [here](https://github.com/compucorp/uk.co.compucorp.civicase/blob/master/CRM/Civicase/Hook/BuildForm/TokenTree.php#L381), because of this, the Email Recipient is not initialized with core tokens and this resulted in unexpected results causing the token category to be displayed as Undefined. To fix this, the condition to check was changed to 
```php
if (empty($newTokenTree[self::RECIPIENT_TOKEN_TEXT]['children'][0])) 
```
Which correctly checks whether the core fields for the token has been added or not. Also if custom field tokens has already been added to the token category, they are re-added back when the token category is initialized with the core fields. 
Same issue could happen for case tokens and similar fix was applied there too. The token data in the tests was re-ordered and more custom fields added so that custom fields come before and after core fields to simulate the scenario for this issue.

- The Address extends entity was added to the list of custom fields to be fetched, this was missing before and was causing some custom fields token label to have an empty value based on the condition [here](https://github.com/compucorp/uk.co.compucorp.civicase/blob/master/CRM/Civicase/Hook/BuildForm/TokenTree.php#L415)

### CODE REVIEW

**Review by reneolivo — APPROVED** [review]

**Comment by jamienovick** [comment]
Why does the SSP version for fontawesome affect CiviCRM screens?

**Review by lisandro-compucorp — APPROVED** [review]
I checked the last two commits (BE code) and it is approved 👍  It is a complicated logic, a little hard to follow (now and more in the future), but it will work.
